tests/cases/py_client/lib_test.py

```diff
@@ -438,6 +438,17 @@ def processEvent(event):
         file = self.client.uploadFileToItem(item['_id'], testPath)
         self.assertEqual(file['mimeType'], 'text/plain')
 
+        # Test uploading to a folder
+        self.client.uploadFileToFolder(
+            str(self.publicFolder['_id']), path, reference='test4_reference')
+        while (not events.daemon.eventQueue.empty() and
+                time.time() - starttime < 5):
+            time.sleep(0.05)
```

Comment:
There's a synchronous event we can use to test this, `'model.file.finalizeUpload.after'`. The event receives both the 'upload' and 'file' documents as keys.